astropy/io/misc/asdf/tags/transform/compound.py

```diff
@@ -42,6 +42,7 @@
 class CompoundType(TransformType):
     name = ['transform/' + x for x in _tag_to_method_mapping.keys()]
     types = [CompoundModel]
+    version = '1.2.0'
```

Comment:
Unfortunately this change means `fix_inputs` is looking for a schema with version number 1.2.0 (only version 1.1.0 is available) and asdf raises a warning before using version 1.1.0. astropy turns the warning into an error. I think the way around this is to catch the warning in the test for now but have an asdf public release with the cleaned up versioning problems soon.

Reply:
I'm not so sure then about the 4.0.1 milestone. Bumping up required dependency versions in bugfix releases is not a thing we should do. Maybe it's possible to split the PR into two then? One to backport, too and one to 4.1 only?